webapp/channels/src/components/user_settings_modal/user_settings_modal.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useRef, useCallback} from 'react';
+import {Modal, type ModalBody} from 'react-bootstrap';
+import {useIntl} from 'react-intl';
+
+import ModalSidebar from 'components/widgets/modals/components/modal_sidebar';
+
+import AdvancedDisplaySettings from './advanced_display_settings';
+import {useUserSettingsTabs} from './utils';
+import './user_settings_modal.scss';
+
+type Props = {
+    onExited: () => void;
+}
+
+const UserSettingsModal = ({onExited}: Props) => {
+    const [activeTab, setActiveTab] = useState('advanced');
+    const [show, setShow] = useState<boolean>(true);
+    const modalBodyRef = useRef<ModalBody>(null);
```

Comment:
I know there's more to be added to this component, but Is this ref actually needed?